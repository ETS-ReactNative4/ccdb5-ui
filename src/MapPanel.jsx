import './MapPanel.less'
import ActionBar from './ActionBar'
import { connect } from 'react-redux'
import DateIntervals from './DateIntervals'
import ErrorBlock from './Error'
import FilterPanel from './FilterPanel'
import FilterPanelToggle from './FilterPanelToggle'
import Loading from './Dialogs/Loading'
import MapToolbar from './MapToolbar'
import { mapWarningDismissed } from './actions/view'
import PerCapita from './PerCapita'
import React from 'react'
import RowChart from './RowChart'
import { Separator } from './Separator'
import StaleDataWarnings from './StaleDataWarnings'
import TileChartMap from './TileChartMap'
import Warning from './Warning'


const WARNING_MESSAGE = '“Complaints per' +
  ' 1,000 population” is not available with your filter selections.'

export class MapPanel extends React.Component {
  // eslint-disable-next-line complexity
  render() {
    return (
      <section className="map-panel">
        <ActionBar/>
        <StaleDataWarnings />
        { this.props.error &&
          <ErrorBlock text="There was a problem executing your search" />
        }
        { this.props.showWarning &&
          <Warning text={ WARNING_MESSAGE }
                   closeFn={this.props.onDismissWarning}/> }
        { this.props.showMobileFilters && <FilterPanel/> }
        <div className="layout-row refine">
          <FilterPanelToggle/>
          <Separator />
          <DateIntervals/>
          <PerCapita/>
        </div>
        <TileChartMap/>
        <MapToolbar/>
        <RowChart aggtype="product" />
        <RowChart aggtype="issue" />
        <Loading isLoading={ this.props.isLoading || false }/>
      </section>
    )
  }
}

const mapStateToProps = state => ( {
  error: state.map.error,
  isLoading: state.map.isLoading,
  showMobileFilters: state.view.width < 750,
  showWarning: !state.query.enablePer1000 && state.query.mapWarningEnabled
} )

export const mapDispatchToProps = dispatch => ( {
  onDismissWarning: () => {
    dispatch( mapWarningDismissed() )
  }
} )


export default connect( mapStateToProps, mapDispatchToProps )( MapPanel )
