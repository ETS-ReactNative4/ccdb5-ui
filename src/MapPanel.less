@import (less) "base.less";

.map-panel {

  h4 {
    border: none;
  }
  .refine {
    section {
      margin: @grid_gutter-width/3 0 @grid_gutter-width/3 @grid_gutter-width;
      p {
        font-size: @size-vi;
        font-weight: 600;
        color: @gray-40;
      }
      button {
        color: @black;
        &.selected {
          font-weight: 600;
        }
      }
    }
  }
}
