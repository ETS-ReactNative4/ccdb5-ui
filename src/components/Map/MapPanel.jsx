import '../RefineBar/RefineBar.less'
import ActionBar from '../ActionBar'
import { connect } from 'react-redux'
import DateRanges from '../RefineBar/DateRanges'
import ErrorBlock from '../Warnings/Error'
import FilterPanel from '../Filters/FilterPanel'
import FilterPanelToggle from '../Filters/FilterPanelToggle'
import Loading from '../Dialogs/Loading'
import MapToolbar from './MapToolbar'
import { mapWarningDismissed } from '../../actions/view'
import PerCapita from '../RefineBar/PerCapita'
import { processBars } from '../../utils/chart'
import React from 'react'
import RowChart from '../Charts/RowChart'
import { Separator } from '../RefineBar/Separator'
import StaleDataWarnings from '../Warnings/StaleDataWarnings'
import TileChartMap from '../Charts/TileChartMap'
import Warning from '../Warnings/Warning'


const WARNING_MESSAGE = '“Complaints per' +
  ' 1,000 population” is not available with your filter selections.'

export class MapPanel extends React.Component {
  // eslint-disable-next-line complexity
  render() {
    return (
      <section className="map-panel">
        <ActionBar/>
        <StaleDataWarnings />
        { this.props.error &&
          <ErrorBlock text="There was a problem executing your search" />
        }
        { this.props.showWarning &&
          <Warning text={ WARNING_MESSAGE }
                   closeFn={this.props.onDismissWarning}/> }
        { this.props.showMobileFilters && <FilterPanel/> }
        <div className="layout-row refine-bar">
          <FilterPanelToggle/>
          <Separator />
          <DateRanges/>
          <PerCapita/>
        </div>
        <TileChartMap/>
        <MapToolbar/>
        <RowChart id="product"
                  colorScheme={this.props.productData.colorScheme}
                  data={this.props.productData.data}
                  title="Product by highest complaint volume"/>
        <RowChart id="issue"
                  colorScheme={this.props.issueData.colorScheme}
                  data={this.props.issueData.data}
                  title="Issue by highest complaint volume"/>
        <Loading isLoading={ this.props.isLoading || false }/>
      </section>
    )
  }
}

const mapStateToProps = state => ( {
  error: state.map.error,
  isLoading: state.map.isLoading,
  issueData: processBars( state.query.issue, state.map.results.issue, false ),
  productData: processBars( state.query.product, state.map.results.product,
    false ),
  showMobileFilters: state.view.width < 750,
  showWarning: !state.query.enablePer1000 && state.query.mapWarningEnabled
} )

export const mapDispatchToProps = dispatch => ( {
  onDismissWarning: () => {
    dispatch( mapWarningDismissed() )
  }
} )


export default connect( mapStateToProps, mapDispatchToProps )( MapPanel )
