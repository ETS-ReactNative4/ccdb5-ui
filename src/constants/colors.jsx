/* eslint-disable no-inline-comments, camelcase */
module.exports = Object.freeze( {
  BriteCharts: {
    regular: '#20aa3f',
    medium: '#ADDC91',
    light: '#C7E5B3'
  },
  DataLens: {
    0: '#2cb34a',
    1: '#addc91',
    2: '#257675',
    3: '#9ec4c3',
    4: '#0072ce',
    5: '#96c4ed',
    6: '#254b87',
    7: '#9daecc',
    8: '#b4267a',
    9: '#dc9cbf',
    10: '#a2a3a4'
  }
} )
