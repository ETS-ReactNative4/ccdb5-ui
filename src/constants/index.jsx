// Internal triggers
export const PAGE_CHANGED              = 'PAGE_CHANGED'
export const SEARCH_CHANGED            = 'SEARCH_CHANGED'
export const SIZE_CHANGED              = 'SIZE_CHANGED'
export const SORT_CHANGED              = 'SORT_CHANGED'
export const URL_CHANGED               = 'URL_CHANGED'
export const FILTER_CHANGED            = 'FILTER_CHANGED'
export const FILTER_REMOVED            = 'FILTER_REMOVED'
export const FILTER_ALL_REMOVED        = 'FILTER_ALL_REMOVED'

// External Triggers
export const COMPLAINTS_RECEIVED       = 'COMPLAINTS_RECEIVED'
export const COMPLAINTS_FAILED         = 'COMPLAINTS_FAILED'

// Useful constants
export const knownFilters = [
  'company',
  'company_public_response',
  'company_response',
  'consumer_consent_provided',
  'consumer_disputed',
  'issue',
  'product',
  'state',
  'submitted_via',
  'tag',
  'timely',
  'zip_code'
]

export const SLUG_SEPARATOR            = 'â€¢'
