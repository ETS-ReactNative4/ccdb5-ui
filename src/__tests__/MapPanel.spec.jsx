import configureMockStore from 'redux-mock-store'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import MapPanel from '../MapPanel'
import { MODE_MAP } from '../constants'
import React from 'react'
import renderer from 'react-test-renderer'
import thunk from 'redux-thunk'

function setupSnapshot( printMode ) {
  const items = [
    { key: 'CA', doc_count: 62519 },
    { key: 'FL', doc_count: 47358 }
  ]

  const middlewares = [ thunk ]
  const mockStore = configureMockStore( middlewares )
  const store = mockStore( {
    aggs: {
      doc_count: 100,
      total: items.length
    },
    map: {
      product: [],
      state: []
    },
    query: {
      from: 0,
      size: 10,
      tab: MODE_MAP,
      product: [
        { name: 'foo' }
      ]
    },
    results: {
      items
    },
    view: {
      printMode
    }
} )

  return renderer.create(
    <Provider store={ store }>
      <IntlProvider locale="en">
        <MapPanel />
      </IntlProvider>
    </Provider>
  )
}

describe( 'component:MapPanel', () => {
  it( 'renders without crashing', () => {
    const printMode = false
    const target = setupSnapshot( printMode )
    const tree = target.toJSON()
    expect( tree ).toMatchSnapshot()
  } )

  it( 'renders Print without crashing', () => {
    const printMode = true
    const target = setupSnapshot( printMode )
    const tree = target.toJSON()
    expect( tree ).toMatchSnapshot()
  } )
} )
