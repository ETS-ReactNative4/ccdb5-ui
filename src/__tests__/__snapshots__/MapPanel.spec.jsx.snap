// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:MapPanel renders printMode without crashing 1`] = `
<section
  className="map-panel"
>
  <section
    className="print-info"
  >
    <p>
      Dates: 
       - 
    </p>
    <p>
      Showing 2 out of 100 total complaints
    </p>
  </section>
  <summary
    className="action-bar"
  >
    <div>
      <h2>
        Showing 
        <span>
          2
        </span>
         matches out of 
        <span>
          100
        </span>
         total complaints
      </h2>
    </div>
    <div
      className="layout-row"
    >
      <h4
        className="flex-all"
      >
        <button
          className="a-btn a-btn__link"
          data-gtm_ignore="true"
          onClick={[Function]}
        >
          Export results
        </button>
      </h4>
    </div>
  </summary>
  <div
    className="layout-row refine"
  >
    <section
      className="date-intervals"
    >
      <p>
        Date range (Click to modify range)
      </p>
      <button
        className="date-selector interval-3m"
        onClick={[Function]}
      >
        3m
      </button>
      <button
        className="date-selector interval-6m"
        onClick={[Function]}
      >
        6m
      </button>
      <button
        className="date-selector interval-1y"
        onClick={[Function]}
      >
        1y
      </button>
      <button
        className="date-selector interval-3y"
        onClick={[Function]}
      >
        3y
      </button>
      <button
        className="date-selector interval-All"
        onClick={[Function]}
      >
        All
      </button>
    </section>
    <section
      className="per-capita"
    >
      <p>
        Map Shading
      </p>
      <select
        onChange={[Function]}
      >
        <option
          value="None"
        >
          Complaints
        </option>
        <option
          value="Per 1000 pop."
        >
          Per capita
        </option>
      </select>
    </section>
  </div>
  <div>
    <div
      className="cfpb-chart"
      data-chart-type="tile_map"
      id="tile-chart-map"
    />
  </div>
  <div
    className="map-toolbar"
  >
    <section
      className="state-heading"
    >
      <span>
        United States of America
      </span>
      <span>
        
      </span>
      
    </section>
    
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Product
       by complaint volume (5 highest counts)
    </h3>
    <div
      id="row-chart-product"
    />
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Issue
       by complaint volume (5 highest counts)
    </h3>
    <div
      id="row-chart-issue"
    />
  </div>
  <section
    className="print-info-footer"
  >
    <p>
      URL: 
      http://localhost/
    </p>
  </section>
</section>
`;

exports[`component:MapPanel renders without crashing 1`] = `
<section
  className="map-panel"
>
  <summary
    className="action-bar"
  >
    <div>
      <h2>
        Showing 
        <span>
          2
        </span>
         matches out of 
        <span>
          100
        </span>
         total complaints
      </h2>
    </div>
    <div
      className="layout-row"
    >
      <h4
        className="flex-all"
      >
        <button
          className="a-btn a-btn__link"
          data-gtm_ignore="true"
          onClick={[Function]}
        >
          Export results
        </button>
      </h4>
    </div>
  </summary>
  <div
    className="layout-row refine"
  >
    <section
      className="date-intervals"
    >
      <p>
        Date range (Click to modify range)
      </p>
      <button
        className="date-selector interval-3m"
        onClick={[Function]}
      >
        3m
      </button>
      <button
        className="date-selector interval-6m"
        onClick={[Function]}
      >
        6m
      </button>
      <button
        className="date-selector interval-1y"
        onClick={[Function]}
      >
        1y
      </button>
      <button
        className="date-selector interval-3y"
        onClick={[Function]}
      >
        3y
      </button>
      <button
        className="date-selector interval-All"
        onClick={[Function]}
      >
        All
      </button>
    </section>
    <section
      className="per-capita"
    >
      <p>
        Map Shading
      </p>
      <select
        onChange={[Function]}
      >
        <option
          value="None"
        >
          Complaints
        </option>
        <option
          value="Per 1000 pop."
        >
          Per capita
        </option>
      </select>
    </section>
  </div>
  <div>
    <div
      className="cfpb-chart"
      data-chart-type="tile_map"
      id="tile-chart-map"
    />
  </div>
  <div
    className="map-toolbar"
  >
    <section
      className="state-heading"
    >
      <span>
        United States of America
      </span>
      <span>
        
      </span>
      
    </section>
    
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Product
       by complaint volume (5 highest counts)
    </h3>
    <div
      id="row-chart-product"
    />
  </div>
  <div
    className="row-chart-section"
  >
    <h3>
      Issue
       by complaint volume (5 highest counts)
    </h3>
    <div
      id="row-chart-issue"
    />
  </div>
</section>
`;
